### ISSUE TITLE
`prefer-starts-ends-with`: Remove `string.match`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#713]
We are checking `string.match(/^a/)`, we should not check it, only `/^a/.test(string)`.

**Comment by sindresorhus** [comment]
Why?

**Comment by fisker** [comment]
I was thinking `string.match` returns an array, and `startsWith` returns a boolean. 

They are total different.

But now I can't found a useful case except checking a string starts with such a simple regex.

```js
const [firstLetterA] = string.match(/^a/); 
// VS
const [firstLetterA] = string.startsWith('a');
```

Different, but who use this.

So, also auto fix `string.match` ?  Just feel like missing something.

**Comment by sindresorhus** [comment]
Hmm. Maybe we should drop it, but add a different rule that ensures that `.match` is not used for conditional checks? Since then `.test` would be more correct.

**Comment by silverwind** [comment]
A rule that reports `regex.exec` or `string.match` used in a conditional/boolean context might nice to have.

**Comment by sindresorhus** [comment]
> A rule that reports string.match or regex.exec used in a conditional/boolean context might nice to have. It should suggest regex.test for both.

https://github.com/sindresorhus/eslint-plugin-unicorn/issues/732

**Comment by fisker** [comment]
[`prefer-regexp-test` rule](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/prefer-regexp-test.md) has been added, `String#match()` should be removed.
